### PR TITLE
Add an s3proxy lambda

### PR DESCRIPTION
**Pull request by dimaryaz**
(Only tested using quilt-lite, not as an actual AWS lambda)

### CODE REVIEW

**Review by nl0 — COMMENTED** [review]
this looks ok, but since it's only used in `quilt3_lite`, maybe it makes sense to implement it as an ASGI app/middleware, so it can be mounted directly, not via lambda harness (and if we need to run it in lambda, we can use smth like `magnum` for that)

File: lambdas/s3proxy/index.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    cors_headers = {
+        'access-control-allow-headers': request.headers.get('access-control-request-headers', ''),
+        'access-control-allow-methods': request.headers.get('access-control-request-method', ''),
+        'access-control-allow-origin': '*',
+        'access-control-max-age': '3000',
+        'access-control-expose-headers': ', '.join([
+            'Content-Length',
+            'Content-Range',
+            'ETag',
+            'x-amz-meta-helium',
```
Comment (sir-sigurd) [inline lambdas/s3proxy/index.py:27]: Do we really need this one?

File: lambdas/s3proxy/index.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+    request_headers = dict(request.headers)
+    request_headers.pop('host', None)  # Correct host header will come from the URL.
+    request_headers.pop('connection', None)  # Let requests handle keep-alive, etc.
+
+    response = session.request(
+        method=request.method,
+        url=url,
+        headers=request_headers,
+    )
```
Comment (sir-sigurd) [inline lambdas/s3proxy/index.py:69]: This seems to not work for upload. Probably we don't need it right now, but might make sense to support it from the beginning.

**Review by sir-sigurd — COMMENTED** [review]
I think its not useful as lambda because of lambda's restrictions on size of request/response.

File: lambdas/s3proxy/index.py (hunk 2 above)
Comment (dimaryaz) [reply lambdas/s3proxy/index.py:69]: Oh right, need the body. I didn't actually test uploads.

**Comment by dimaryaz** [comment]
I'll move this into quilt-lite directly, instead of making it a lambda.

**Comment by sir-sigurd** [comment]
Should we close this?

**Pull request closed without merging**